src/test/java/de/tum/in/www1/artemis/ResultServiceIntegrationTest.java

```diff
@@ -294,23 +289,66 @@ public void createManualProgrammingExerciseResult() throws Exception {
         assertThat(response.getFeedbacks().size()).isEqualTo(result.getFeedbacks().size());
     }
 
+    @Test
+    @WithMockUser(value = "tutor1", roles = "TA")
+    public void createManualProgrammingExerciseResult_manualResultsNotAllowed() throws Exception {
+        var participation = setParticipationForProgrammingExercise(AssessmentType.AUTOMATIC);
+        Result result = ModelFactory.generateResult(true, 200).resultString("Good effort!");
+        List<Feedback> feedbacks = ModelFactory.generateFeedback().stream().peek(feedback -> feedback.setText("Good work here")).collect(Collectors.toList());
+        result.setFeedbacks(feedbacks);
+        result.setParticipation(participation);
+
+        String dummyHash = "9b3a9bd71a0d80e5bbc42204c319ed3d1d4f0d6d";
+        when(gitService.getLastCommitHash(ArgumentMatchers.any())).thenReturn(ObjectId.fromString(dummyHash));
+
+        request.postWithResponseBody("/api/participations/" + participation.getId() + "/manual-results", result, Result.class, HttpStatus.FORBIDDEN);
+    }
+
+    @Test
+    @WithMockUser(value = "tutor1", roles = "TA")
+    public void createManualProgrammingExerciseResult_resultExists() throws Exception {
+        var participation = setParticipationForProgrammingExercise(AssessmentType.SEMI_AUTOMATIC);
+        Result result = ModelFactory.generateResult(true, 200).resultString("Good effort!");
+        List<Feedback> feedbacks = ModelFactory.generateFeedback().stream().peek(feedback -> feedback.setText("Good work here")).collect(Collectors.toList());
+        result.setFeedbacks(feedbacks);
+        result.setParticipation(participation);
+        result = resultRepository.save(result);
+
+        String dummyHash = "9b3a9bd71a0d80e5bbc42204c319ed3d1d4f0d6d";
```

Comment:
if you mock this multiple times, please move it into the method
    @BeforeEach
    public void init() {}

Reply:
True. I moved the part of creating result and mocking with dummy hash into the `@BeforeEach` method